### DISCUSSION TITLE
hsrb_command update_release 22.10 fails

### DISCUSSION THREAD

**Discussion by lexihaberl** [HumanSupportRobot/QandA#39] [Q&A] [answered]
Hello,

while trying to update our robot from version 21.12 to version 22.10 we unfortunately got the following error when executing ```hsrb_command update_release 22.10```:

**Error response from daemon: Head "https://docker.hsr.io/v2/hsrb/robot/manifests/noetic": unauthorized: authentication required 
Failed to update docker image.**

The docker registry config.json file does exist for the administrator-user. Is it possible that our config.json file has an older, deprecated authentication key, or is the issue something else? If yes, how could we proceed to update the robot? We would prefer not having to make a clean install.

**Comment by TMats** [comment]
The same problem happened to our HSR too.

**Comment by hsr-admin** [comment, marked as answer]
We have confirmed that the docker pull process causes an error.
It may take time to fix, if you are in a hurry, please use a clean install.

(Reference)
https://docs.hsr.io/hsrb_user_manual_en/howto/clean_install.html#id1

**Reply by hsr-admin** [reply]
We apologize for the inconvenience. The investigation has been difficult and time consuming.

**Reply by hsr-admin** [reply]
The problem with docker pull authentication error has been resolved and confirmed to work properly in 22.04 environment.

**Reply by lexihaberl** [reply]
Thank you for fixing this issue.
I will test whether it works on our robot on friday.

**Reply by lexihaberl** [reply]
I can confirm that it also worked on our side. We were able to update the robot from 21.12 to 22.10.
Thanks again for fixing this issue!
